collector/collector.go

```diff
@@ -145,6 +146,11 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) (up float64) {
 		return 0
 	}
 
+	// Collect and export system HA licenses metrics
+	if err := e.exportSystemHALicenses(ch); err != nil {
+		return 0
+	}
+	
```

Comment:
Yes, I checked this for a non-ha setup of Artifactory OSS in a local kind cluster. In this case, the `/licenses` endpoint response is identical with that of `/license` and the exporter only provides the "license" metric. 

This is caused by the response not being a JSON list of licenses and thus the loop in https://github.com/peimanja/artifactory_exporter/pull/153/files#diff-a09fbd1e8c1ccc91b9a4049c2cbaa31bda227986e9b3f636b8e36fd3645700ddR96 has no iteration.

Debug logs in the exporter:

```
time=2024-12-18T20:20:35.558+01:00 level=DEBUG msg="Fetching license stats"
time=2024-12-18T20:20:35.558+01:00 level=DEBUG msg="Fetching http" path=http://localhost:8081/artifactory/api/system/license
time=2024-12-18T20:20:35.562+01:00 level=DEBUG msg="Fetching HA licenses stats"
time=2024-12-18T20:20:35.562+01:00 level=DEBUG msg="Fetching http" path=http://localhost:8081/artifactory/api/system/licenses
```

Metric output:

```
# HELP artifactory_system_license License type and expiry as labels, seconds to expiration as value
# TYPE artifactory_system_license gauge
artifactory_system_license{expires="",licensed_to="",node_id="artifactory-0",type="oss"} 0

```

We could add a log message on e.g. debug level, which notifies that the HA license metric will not be created, since it is not a HA setup. What do you think?

Reply:
Thanks for checking, if there is no error message after `time=2024-12-18T20:20:35.562+01:00 level=DEBUG msg="Fetching http" path=http://localhost:8081/artifactory/api/system/licenses` we should be good